src/values/ObjectValue.js

```diff
@@ -290,6 +299,19 @@ export default class ObjectValue extends ConcreteValue {
     return this._isPartial.value;
   }
 
+  makeTainted(): void {
+    this._isTainted = this.$Realm.intrinsics.true;
+    // Clear all properties. This object is now completely unknown.
+    this.properties.clear(); // TODO: Track that this happens so that it can be undone by evaluateForEffects
```

Comment:
This is an important TODO that should be addressed before landing.

Reply:
Yup.